core/src/cpu/air/mod.rs

```diff
@@ -230,9 +255,21 @@ impl CpuChip {
         builder
             .when_transition()
             .when(next.is_real)
-            .when(local.selectors.is_jal + local.selectors.is_jalr)
+            .when(is_jump_instruction.clone())
             .assert_eq(jump_columns.next_pc.reduce::<AB>(), next.pc);
 
+        // When is_jump_instruction is true, assert that local.is_real is true.
+        builder
+            .when(is_jump_instruction.clone())
+            .assert_one(local.is_real);
```

Comment:
Do we need the checks like this one? and for is_branching, is_sequential_instr. Seems like the constraints on is_real by itself (first row is real, once not real can't become real again) should be enough.

Reply:
ya, agreed.